pkg/debugapi/peer.go

```diff
@@ -35,6 +35,11 @@ func (s *server) peerConnectHandler(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	s.Addressbook.Put(address, addr)
+	if err := s.TopologyDriver.AddPeer(r.Context(), address); err != nil {
+		s.Logger.Debugf("debug api: topologyDriver.AddPeer %s: %v", addr, err)
```

Comment:
why doesn't the API return an error in this case?

Reply:
Oh, I thought you wanted it to be debug level? Error level message is the one visible to the user and I left it as it was. But then on the second thought I thought that it should not be an error, or we should cleanup and disconnect in this case. I don't know if we wanna go there at this point. LMNWYT? @janos as well as he was working on `debugapi`?

Reply:
Basically, should we fail in this case, and how should we handle it. I am not sure what should be the best behaviour in the case of debug API.

Reply:
I meant the logging should be done on the debug level, but I think it would be nice to know if the requested debug api call failed for whatever reason. In this case it would just fail silently and indicate nothing both to the debug api caller nor to the user (unless the user has logging in debug level on).
We should be just careful about how we use the logger, with the understanding on who is reading which log levels, that is all. I did not mean that the api should not fail in this case

Reply:
This certainly need to be responded with the 500, and I am not sure why the error call is removed (conversation above). I think that it was just right initially.

Reply:
@janos, originally it was logging both on `error` and `debug` loglevels, this was changed here https://github.com/ethersphere/bee/pull/26/commits/a2034d29d5655cbe943636c73f8ca8c8e89fb470. i claim we should only log on debug (since this is a developer message, not message intended for the end user), and return a `500`, that is all.

Reply:
My point is, that if we return an error message in this case, like it was in the first place, the user would still be connected. Should we disconnect in this case? 
@acud I think that the pattern is:
Debug: message for the developer
Error: user facing message

That is why there was an detailed message in the debug, and more context based - connect failed in the error message

Reply:
Yes, I know, I think that both lines should be there, so that error message is detectable when this error happens if only error log level severity is set.

Reply:
So, no need to disconnect and stuff, just return an error to do user? I think that connect will still work in this case.

Reply:
I'm not sure what you mean by `user would still be connected` and what do you mean by disconnecting... if the `AddPeer` call failed I would not expect the peer to be connected and of the underlying data structures to mutate, so appropriate cleanups and rollbacks should be done. It might be possible to do a cleanup on the address book entry (which cannot be used), but that is optional

Reply:
If you take a look at the code above, we have a `p2p.Connect` call, and then we add stuff to the addressbook and add to the topology driver. So my question is, if adding to the topology driver fails, what should happen? Should we disconnect this peer? I am not sure how we should handle this 100% correct.

Reply:
aha, yes, should disconnect.

Reply:
Done :)